File: Algorithms/Strings/Caesar Cipher.js
// https://www.hackerrank.com/challenges/caesar-cipher-1/problem

function caesarCipher(s, k) {
  const alphabet = 'abcdefghijklmnopqrstuvwxyz'
  k = k % alphabet.length
  let res = ''
  const cipher = alphabet.slice(k) + alphabet.slice(0, k)

  for (let char of s) {
    if (isAlphabet(char)) {
      if (isUpper(char)) {
        const temp = char.toLowerCase()
        const idx = alphabet.indexOf(temp)
        res += cipher[idx].toUpperCase()
      } else {
        const idx = alphabet.indexOf(char)
        res += cipher[idx]
      }
    } else {
      res += char
    }
  }

  return res
}

File: Algorithms/Strings/Caesar Cipher.py
# Caesar Cipher

# 1
def caesarCipher(s, k):
  # Write your code here
  k = k % 26
  res = ''

  for i in range(len(s)):
    if s[i].isupper():
      code = ord(s[i]) + k
      if code > ord('Z'):
        correct_code = ord('A') + (code - ord('Z') - 1)
        res += chr(correct_code)
      else:
        res += chr(code)
    elif s[i].islower():
      code = ord(s[i]) + k
      if code > ord('z'):
        correct_code = ord('a') + (code - ord('z')- 1)
        res += chr(correct_code)
      else:
        res += chr(code)
    else:
      res += s[i]
  return res